Add note about HMR WebSocket to upgrade guide (#30704)
vercel/next.js: docs/upgrading.md

See the [documentation](https://nextjs.org/docs/basic-features/image-optimization#styling) for more info.

### Next.js' HMR connection now uses a WebSocket

> Previously Next.js used a [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) connection to receive HMR events but now uses a WebSocket connection.

Reviewer: ```suggestion
Previously, Next.js used a [server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) connection to receive HMR events. Next.js 12 now uses a WebSocket connection.
```

> In some cases when proxying requests to Next.js' dev server you will need to ensure the upgrade request is handled correctly, for example, in `nginx` you would need to add the below configuration:

Reviewer: ```suggestion
In some cases when proxying requests to the Next.js dev server, you will need to ensure the upgrade request is handled correctly. For example, in `nginx` you would need to add the following configuration:
```

> ```

Reviewer: Do we want to add any highlighting? IIRC it looks strange if you don't specify any on .org

Author: Looks like nginx is a supported type so added that 👍

> ```

Author: ```suggestion
```nginx
```
